smithy-aws-protocol-tests/model/restXml/http-query.smithy

```diff
@@ -351,3 +363,70 @@ structure QueryIdempotencyTokenAutoFillInput {
     @idempotencyToken
     token: String,
 }
+
+// Named query members take precedence over unnamed.
+@http(uri: "/Precedence", method: "POST")
+operation QueryPrecedence {
+    input: QueryPrecedenceInput
+}
+
+apply QueryPrecedence @httpRequestTests([
```

Comment:
Added in 7b30e05.

Should there also be specific tests for precedence when targeting `Map<String, List<String>>` or is it clear enough how that would work?

Reply:
If you're asking whether to add a service test -- yes a service test would be nice. I would expect the precedence to not matter since a map slurps up all query string parameters, meaning some will be duplicated in specific httpQuery bindings and httpQueryParam bindings

Reply:
I added a service test that targets `Map<String, String>`.  I don't think an additional service test that covers `Map<String, List<String>>` is needed, but wanted to call that out in case others felt that level of specificity would be useful.

Reply:
I do think it's needed. We should be comprehensive in the tests to make sure everything gets implemented without needing to consult the spec.

Reply:
I added additional tests and updated the existing tests to cover server and client for mixed `httpQuery` and `httpQueryParams` bindings that target Map<String, String> and Map<String, List<String>>.